### PR TITLE
Directory operations and Preserve SMB Attributes

### PR DESCRIPTION
**Pull request by EmmaZhu**
1.  Upgraded XSCL from 10.0.3 to 11.1.0
2.  Fixed an issue of CPU is still in using when there's nothing being transferred in DataMovement Library
3.  Fixed an issue of reporting exception when running on a specific environment with FIPS enabled
4. Optimized on Azure File directory creation when uploading from local file directory to Azure File directory and copying from Azure Blob virtual directory to Azure File directory
5. Added support to copying empty directories when uploading/downloading from/to local file directory to/from Azure File directory
 6. Added support to preserve SMB attributes when uploading/downloading from/to local file directory to/from Azure File directory on Windows. SMB attributes includes creation time, last write time and NTFS File Attributes.

### CODE REVIEW

File: lib/TransferJobs/SubDirectoryTransfer.cs [hunk 1]
```diff
@@ -350,11 +350,19 @@ private void CreateDestinationDirectory(CancellationToken cancellationToken)
                 }
                 catch (StorageException storageException)
                 {
-                    throw new TransferException(TransferErrorCode.FailToVadlidateDestination,
-                        string.Format(CultureInfo.CurrentCulture,
-                            Resources.FailedToValidateDestinationException,
-                            storageException.ToErrorDetail()),
-                        storageException);
+                    if ((null != storageException.InnerException)
```
Comment (blueww) [inline lib/TransferJobs/SubDirectoryTransfer.cs]: Would you like to look at other place to see it the OperationCanceledException also wrappered to StorageException by XSCL.

Comment (blueww) [inline lib/TransferJobs/SubDirectoryTransfer.cs]: Would you like to see if other place will wrapper OperationCanceledException to storageException in cancel?

Comment (EmmaZhu) [reply lib/TransferJobs/SubDirectoryTransfer.cs]: The change here is for our own exception handling.
In HierarchyDirectoryTransfer, for it lists source directory and its subdirectories with multiple threads, when there's any error happens in any one list threading, DMLib will try to stop the whole HierarchyDirectoryTransfer with cancelling, throw out the origin exception which causes the error. Thus DMLib would need to ignore OperationCancelled exception (which is caused by ourselves), and throw out other StorageException (which may represent the actual error root cause).

Comment (EmmaZhu) [reply lib/TransferJobs/SubDirectoryTransfer.cs]: For StorageException needs to thrown out in other place, DMLib's current behavior is to directly throw it out without handling. So we may not need to change it.

**Pull request merged**